### ISSUE TITLE
Investigate if embeddings lookup with `tf.RaggedTensor` is slower than with `tf.Tensor` with latter versions of TF

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/models#1038]

**Comment by gabrielspmoreira** [comment]
Comment by @vysarge 

> a recent test on my side suggests using tf.keras.layers.Embedding + reduce_mean with RaggedTensor input remains slower than with dense Tensor input as of TF 2.12.0 (tensorflow/tensorflow:2.12.0-gpu container).
RaggedTensors in these tests were constructed using RaggedTensor.from_tensor with the padding argument set.  This means ultimately these tensors were constructed with [from_row_splits](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.tensorflow.org%2Fapi_docs%2Fpython%2Ftf%2FRaggedTensor%23from_row_splits&data=05%7C01%7Cgmoreira%40nvidia.com%7C3afd25aa5239472a885108db2c9f9cb6%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638152836278046579%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=orUzrYwPLGVk4opChZN8zgQNXCRkxXvhPwKLYaAG5s4%3D&reserved=0) and that value_rowids and row_lengths indexing schemes are not cached for these tensors (so some ops can be slower than for a RaggedTensor constructed with [from_value_rowids](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.tensorflow.org%2Fapi_docs%2Fpython%2Ftf%2FRaggedTensor%23from_value_rowids&data=05%7C01%7Cgmoreira%40nvidia.com%7C3afd25aa5239472a885108db2c9f9cb6%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638152836278202803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=FXYAED7H0qjbcxDpld43H9olaKjykV5BN1ecEgjonR8%3D&reserved=0), for example).
A confounding factor is that many of these tests are currently slower overall due to the switch from legacy to experimental optimizer code.  For example, one SGD test case measured [here](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fdocs.google.com%2Fspreadsheets%2Fd%2F1jlKDVeoMvpQfyCF9RFmR3VxckbPbrvBg2POwbF2p7RM%2Fedit%23gid%3D135622185&data=05%7C01%7Cgmoreira%40nvidia.com%7C3afd25aa5239472a885108db2c9f9cb6%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638152836278202803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=r%2FHAehngjWVkvy8rXZ7Kv7mUU4VOtgzu3gOVatL%2FnLw%3D&reserved=0) as 10ms/step with dense Tensor and 47ms/step with RaggedTensor now appears to take 35ms/step with dense Tensor and 80ms/step with RaggedTensor.
See also: [nvbugs/3908345](https://nam11.safelinks.protection.outlook.com/?url=https%3A%2F%2Fnvbugswb.nvidia.com%2FNvBugs5%2FSWBug.aspx%3Fbugid%3D3908345&data=05%7C01%7Cgmoreira%40nvidia.com%7C3afd25aa5239472a885108db2c9f9cb6%7C43083d15727340c1b7db39efd9ccc17a%7C0%7C0%7C638152836278202803%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=DPMIP%2FrXS8j0lz9qt%2FLFSvNduae6ruwChpoGvp69mfs%3D&reserved=0)
I did a very small test only, so it may not be the same for embeddings with other dimensions etc.
